debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/converters/sink/types/bson/BytesType.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.converters.sink.types.bson;
+
+import java.nio.ByteBuffer;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.errors.DataException;
+import org.bson.BsonBinary;
+import org.bson.BsonValue;
+
+public class BytesType extends AbstractBsonType {
+
+    public BytesType() {
+        super(Schema.BYTES_SCHEMA);
+    }
+
+    @Override
+    public BsonValue toBson(final Object data) {
+        if (data instanceof ByteBuffer) {
+            return new BsonBinary(((ByteBuffer) data).array());
+        }
+        if (data instanceof byte[]) {
+            return new BsonBinary((byte[]) data);
+        }
```

Comment:
To be consistent, lets use `if/elseif/else`.

Reply:
Why though? if-elseif-else makes sense in case of branching statement with consecutive code to be executed. When it's essentially an expression switch based on types (which should be used here once we have java 17) this seems more readable.

Reply:
OK, let's make a decision. It is not sure we'll get this into 2.7. Even if we do we already have ibmi connector requiring Java 17. So I propose to change all the code to use switch/pattern matching. It will be kind of prototype for us for to build a code style around it.

Reply:
Hold on, this is not possible with 2.7 as it is the same codebase with MongoDB source connector :-(